en/orm/retrieving-data-and-resultsets.rst

```diff
@@ -758,6 +758,18 @@ associations::
 
 Again, the only difference is that no additional columns will be added to the
 result set, and no ``_matchingData`` property will be set.
+However, it is possible to combine ``innerJoinWith()`` and ``contain()`` when you need to filter by associate data and you want also to retrieve associate fields too (following the same filter)::
+
+    $filter = ['Tags.name' => 'CakePHP'];
+    $query = $articles->find();
+    $query->distinct($articles);
+    $query->contain('Tags', function (Cake\ORM\Query $q) use ($filter) {
+        return $q->where($filter);
+    })
+          ->innerJoinWith('Tags', function (Cake\ORM\Query $q) {
```

Comment:
Use `\Cake\ORM\Query`, because it will resolve to wrong namespace when in another namespace.